### PR TITLE
caf: events: Add documentation for hardware-related events

### PR DESCRIPTION
**Pull request by MarekPieta**
Change adds doxygen documentation for button_event, click_event and led_event.

Jira: NCSDK-9551

### CODE REVIEW

File: include/caf/events/led_event.h [hunk 1]
```diff
@@ -20,22 +20,40 @@
 extern "C" {
 #endif
 
-/** @brief LED event used to change LED operating mode and color. */
+/** @brief LED event.
```
Comment (greg-fer) [inline include/caf/events/led_event.h:23]: There's no need for such a header in brief. Just write the brief from lines 25-27 here.

(Global edit for all files)

File: include/caf/events/led_event.h [hunk 2]
```diff
@@ -20,22 +20,40 @@
 extern "C" {
```
Comment (greg-fer) [inline include/caf/events/led_event.h:20]: For the defgroup in lines 10-14, please follow the guidelines here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/doc_styleguide.html#file-headers-and-groups

(Global edit for all files)

File: include/caf/events/led_event.h [hunk 3]
```diff
@@ -20,22 +20,40 @@
...
 
-/** @brief LED ready event used to notify that leds are ready for next
- * LED stream effect.
+/** @brief LED ready event.
+ *
+ * The LED ready event is submitted to inform that a LED has finished displaying a LED effect.
+ * The event is used to notify that the LED is ready to display next effect.
+ *
+ * If the displayed LED effect has loop_forever set to true, the LED ready event should not be
+ * submitted (the effect never ends).
```
Comment (greg-fer) [inline include/caf/events/led_event.h:47]:
```suggestion
If the displayed LED effect has loop_forever set to true, the effect never ends and the LED ready event should not be submitted.
```

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: include/caf/events/led_event.h (hunk 1 above)
Comment (MarekPieta) [reply include/caf/events/led_event.h:23]: Discussed offline.
The doc style guide does not force us either to add or not to add brief. In most places we used to add brief and include details in separate paragraph if needed. Maybe we could stick to this convention?

File: include/caf/events/led_event.h (hunk 2 above)
Comment (MarekPieta) [reply include/caf/events/led_event.h:20]: fixed

**Review by greg-fer — APPROVED** [review]

**Review by pdunaj — APPROVED** [review]

**Pull request merged**